pkg/controller/enterprisesearch/enterprisesearch_controller.go

```diff
@@ -363,7 +358,7 @@ func watchEsTLSCertsSecret(ent entv1beta1.EnterpriseSearch, watched watches.Dyna
 		return nil
 	}
 	return watched.Secrets.AddHandler(watches.NamedWatch{
-		Name:    tlsSecretWatchName(ent),
+		Name:    tlsSecretWatchName(k8s.ExtractNamespacedName(&ent)),
```

Comment:
It does not come from this PR but I'm wondering if we really need this watch since the `Secret` that contains the Elasticsearch CA is owned by the `entsearch` resource ? Owned resources are already watched: https://github.com/elastic/cloud-on-k8s/blob/883f45d16f8f5f4f09474fb2ada0cafac33d797c/pkg/controller/enterprisesearch/enterprisesearch_controller.go#L118-L124

Reply:
Yes! Let's remove it.